consensus/qkchash/llrb.go

```diff
@@ -0,0 +1,457 @@
+package qkchash
+
+const (
+	Null_Value = uint64(0xFFFFFFFFFFFFFFFF)
+)
+
+// Tree is a Left-Leaning Red-Black (LLRB) implementation of 2-3 trees
+type LLRB struct {
```

Comment:
nit: seems those data structures don't need be exported

Reply:
also, seems you are using https://github.com/petar/GoLLRB/blob/master/llrb/llrb.go, does its license allow this usage without attribution?

Reply:
updated, just keep functions we need, and those functions' implementation is similar to cpp implementation.